hathor/transaction/storage/transaction_storage.py

```diff
@@ -599,11 +595,20 @@ def _topological_sort_metadata(self) -> Iterator[BaseTransaction]:
         raise NotImplementedError
 
     @abstractmethod
-    def add_to_indexes(self, tx: BaseTransaction) -> None:
+    def add_to_indexes_complete(self, tx: BaseTransaction) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def add_to_indexes_pre_consensus(self, tx: BaseTransaction) -> None:
```

Comment:
Missing docstring. How is this method different from `post_consensus(...)`? And why it matters for the indexes?

Shouldn't the consensus algorithm "notify" the indexes that a transaction has been marked as voided or unmarked as voided, so the indexes can update accordingly?

Reply:
If I understood correctly, for the index to work fine when a tx is marked as voided, you must guarantee that in your example with `tx_A` and `tx_B` in conflict, you will never have `add(tx_A) add(tx_B) void(tx_A)`, right? You must guarantee that you are deleting `tx_A` before adding `tx_B`.

Is this what the indexes add split is doing?

Reply:
Great! Thanks!